packages/repository/src/decorators/repository.ts

```diff
@@ -30,7 +30,7 @@ export class RepositoryMetadata {
   /**
    * Class of the model
    */
-  modelClass?: typeof juggler.PersistedModel | typeof Model;
+  modelClass?: typeof Entity;
```

Comment:
`Model` is not enough, we need `Entity`. The reason is that `DefaultCrudRepository` created by the decorator requires `Entity` and its id-related API.

Reply:
Please note that `@repository` should be able to indicate any type of repository, not only `DefaultCrudRepository`.

Reply:
That makes a lot of sense! Unfortunately the current implementation does not support any repository type AFAICT, because it always creates an instance of `DefaultCrudRepository`. See https://github.com/strongloop/loopback-next/blob/e8721422e9a6ea3535e5da75691c464a8fa5cfa5/packages/repository/src/decorators/repository.ts#L140-L143

IMO, this pull request is not making the current situation any worse than it is, therefore we should leave support for arbitrary repository out of scope.